src/shared/sam/activation.ts

```diff
@@ -287,6 +288,33 @@ async function createYamlExtensionPrompt(): Promise<void> {
             yamlPromptDisposables
         )
 
+        /**
+         * Prompt the user to install the YAML plugin when AWSTemplateFormatVersion becomes available as a top level key
+         * in the document
+         * @param event An vscode text document change event
+         * @returns nothing
+         */
+        function promptOnAWSTemplateFormatVersion(event: vscode.TextDocumentChangeEvent): void {
+            for (let line = 0; line < event.document.lineCount; line++) {
```

Comment:
Is there any perf impact for big YAML files? This is being ran on every doc change so I'd at least use `contentChanges` instead of iterating over the whole document every time (does Cloud9 give us `contentChanges` ??)

Reply:
It worked upto 40k lines fine but I switched it to just look at the changed line which should work even faster